contracts/delegatecall/LibEtherForENS.sol

```diff
@@ -0,0 +1,25 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+pragma solidity ^0.8.20;
+
+abstract contract Resolver {
+    function addr(bytes32 node) public view virtual returns (address);
+}
+
+abstract contract ENS {
+    function resolver(bytes32 node) public view virtual returns (Resolver);
+}
+
+library LibEtherForENS {
+    function etherForENS(bytes32 node, uint256 value, bytes memory payload) external{
+        // address of The ENS registry
+        ENS ens = ENS(0x00000000000C2E074eC69A0dFb2997BA6C7d2e1e);
+
+        Resolver resolver = ens.resolver(node);
+        address resolution = resolver.addr(node);
+
+        (bool success, bytes memory returndata) = resolution.call{value: value}(payload);
+        require(success, string(returndata));
```

Comment:
Please prefer using the following function instead for propagating errors.

https://github.com/cartesi/rollups-contracts/blob/2ab9e2388eed4f9f56b0cd2a45be320d5d3e1001/contracts/library/LibError.sol#L9